VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -154,109 +110,156 @@ def configure_waagent(self):
         # todo: restart waagent if necessary to ensure changes are picked up?
         return True
 
-    def configure_fstab(self):
+    def _configure_fstab(self):
         """ remove resource disk from /etc/fstab if present """
         cmd = "sed -i.bak '/azure_resource-part1/d' /etc/fstab"
         if self.executor.ExecuteInBash(cmd) != CommonVariables.process_success:
             self.logger.log(msg="Failed to configure resource disk entry of /etc/fstab", level=CommonVariables.WarningLevel)
-            return False        
+            return False
         return True
 
-    def unmount_resource_disk(self):
+    def _unmount_resource_disk(self):
         """ unmount resource disk """
-        # after service healing multiple unmounts of key file mount point may be required
-        self.disk_util.umount(self.RD_KEY_FILE_MOUNT_POINT)
-        self.disk_util.umount(self.RD_KEY_FILE_MOUNT_POINT)
         self.disk_util.umount(self.RD_MOUNT_POINT)
         self.disk_util.umount('/mnt')
 
-    def is_crypt_mounted(self):
+    def _is_plain_mounted(self):
+        """ return true if mount point is mounted from a non-crypt layer """
+        mount_items = self.disk_util.get_mount_items()
+        for mount_item in mount_items:
+            if mount_item["dest"] == self.RD_MOUNT_POINT and not (mount_item["src"].startswith(CommonVariables.dev_mapper_root) or mount_item["src"].startswith(self.DEV_DM_PREFIX)):
+                return True
+        return False
+
+    def _is_crypt_mounted(self):
         """ return true if mount point is already on a crypt layer """
         mount_items = self.disk_util.get_mount_items()
         for mount_item in mount_items:
-            if mount_item["dest"] == self.RD_MOUNT_POINT and mount_item["src"].startswith(self.DM_PREFIX):
+            if mount_item["dest"] == self.RD_MOUNT_POINT and (mount_item["src"].startswith(CommonVariables.dev_mapper_root) or mount_item["src"].startswith(self.DEV_DM_PREFIX)):
                 return True
         return False
 
-    def get_rd_device_mapper(self):
-        """ retrieve current device mapper path backing the encrypted resource disk mount point """
+    def _get_rd_device_mappers(self):
+        """
+        Retreive any device mapper device on the resource disk (e.g. /dev/dm-0).
+        Can't imagine why there would be multiple device mappers here, but doesn't hurt to handle the case
```

Comment:
tested against RAID scenarios?

Reply:
Do you mean set up a RAID volume on the resource disk and then see how the code reacts?

Reply:
Yes just a check for any regressions in the case where RAID is set up with a persistent device name scheme similar to /dev/dm-0 .. some now deprecated ARM template demos previously suggested RAID (they were tied to the legacy encrypt format operation).